### PR TITLE
Bugs/Features

### PR DESCRIPTION
**Pull request by kvdlxne**
Fixes CCBlueX/LiquidBounce1.8-Issues#3846 
Fixes CCBlueX/LiquidBounce1.8-Issues#3725
Closes CCBlueX/LiquidBounce1.8-Issues#2177
Closes CCBlueX/LiquidBounce1.8-Issues#3840

### CODE REVIEW

**Comment by superblaubeere27** [comment]
Development is paused until the Port of LiquidBounce is done. I will review (and merge) this one when the cross_version branch is merged.

File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/misc/RichPresence.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import kotlin.concurrent.thread
+
+@ModuleInfo(name = "RichPresence", description = "Show discord presence", category = ModuleCategory.MISC)
+class RichPresence : Module() {
+
+    override fun onToggle(state: Boolean) {
+        val rpc = LiquidBounce.clientRichPresence
+        when (state) {
+            false -> rpc.shutdown()
+            true -> thread { rpc.setup() }
```
Comment (superblaubeere27) [inline 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/misc/RichPresence.kt:16]: Errorhandling is missing.

File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/misc/RichPresence.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+package net.ccbluex.liquidbounce.features.module.modules.misc
+
+import net.ccbluex.liquidbounce.LiquidBounce
+import net.ccbluex.liquidbounce.features.module.Module
+import net.ccbluex.liquidbounce.features.module.ModuleCategory
+import net.ccbluex.liquidbounce.features.module.ModuleInfo
+import net.ccbluex.liquidbounce.utils.ClientUtils
+import kotlin.concurrent.thread
+
+@ModuleInfo(name = "RichPresence", description = "Show discord presence", category = ModuleCategory.MISC)
```
Comment (SenkJu) [inline 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/misc/RichPresence.kt:15]: I don't like the idea of rich presence being a module. Their should be a menu for it somewhere else.

File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/player/AntiLiquidPush.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+ * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
+ * https://github.com/CCBlueX/LiquidBounce/
+ */
+package net.ccbluex.liquidbounce.features.module.modules.player
+
+import net.ccbluex.liquidbounce.features.module.Module
+import net.ccbluex.liquidbounce.features.module.ModuleCategory
+import net.ccbluex.liquidbounce.features.module.ModuleInfo
+
+@ModuleInfo(name = "AntiLiquidPush", description = "Disable liquid pushing", category = ModuleCategory.PLAYER)
```
Comment (SenkJu) [inline 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/player/AntiLiquidPush.kt:12]: Could be turned into a NoSlow value instead.

File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/misc/RichPresence.kt (hunk 2 above)
Comment (superblaubeere27) [reply 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/misc/RichPresence.kt:15]: The question is: Where?

Comment (derech1e) [reply 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/misc/RichPresence.kt:15]: Maybe in the Main menu under Mods?

**Comment by kawaiinekololi** [comment]
I'll try to update it tomorrow for cross version, so I can merge it.

**Pull request merged**